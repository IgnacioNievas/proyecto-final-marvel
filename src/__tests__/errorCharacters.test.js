import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ErrCharacters from '../components/errorescomponent/errorcharacters';

test('<ErrCharacters /> , prueba de existencia de elementos', () => {
	render(<ErrCharacters />);

	const h1 = screen.getByTestId('h1');
	expect(h1.tagName).toBe('H1');
	expect(h1.tagName).not.toBe('SPAN', 'H2', 'H3', 'H4', 'P');
	expect(h1).toHaveTextContent('Personaje no encontrado');
	expect(h1).toBeInTheDocument();

	const h2 = screen.getByTestId('h2');
	expect(h2.tagName).toBe('H2');
	expect(h2.tagName).not.toBe('SPAN', 'H1', 'H3', 'H4', 'P', 'INPUT');
	expect(h2).toHaveTextContent('Tal vez lleve - como por ejemplo : Frog-Man');
	expect(h2).toBeInTheDocument();

	const h3 = screen.getByTestId('h3');
	expect(h3.tagName).toBe('H3');
	expect(h3.tagName).not.toBe('SPAN', 'H1', 'H2', 'H4', 'P', 'INPUT');
	expect(h3).toHaveTextContent('o tal vez no exista el personaje');
	expect(h3).toBeInTheDocument();

	const h4 = screen.getByTestId('h4');
	expect(h4.tagName).toBe('H4');
	expect(h4.tagName).not.toBe('SPAN', 'H1', 'H2', 'H3', 'P', 'INPUT');
	expect(h4).toHaveTextContent(
		'o tal vez no alla elegido ningun personaje favorito tampoco'
	);
	expect(h4).toBeInTheDocument();

	const p = screen.getByTestId('p');
	expect(p.tagName).toBe('P');
	expect(p.tagName).not.toBe('SPAN', 'H1', 'H2', 'H3', 'H4', 'INPUT');
	expect(p).toHaveTextContent('Revise y vuelva a intentarlo por favor');
	expect(p).toBeInTheDocument();
	const erro = renderer.create(<ErrCharacters />).toJSON();

	expect(erro).toMatchSnapshot();
});
