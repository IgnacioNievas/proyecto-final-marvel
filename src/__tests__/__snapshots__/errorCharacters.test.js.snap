// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrCharacters /> , prueba de existencia de elementos 1`] = `
<div
  className="sc-bdnxRM vYBlF"
>
  <h1
    data-testid="h1"
  >
    Personaje no encontrado
    <i
      className="fas fa-exclamation-triangle"
    />
  </h1>
  <h2
    data-testid="h2"
  >
     Tal vez lleve - como por ejemplo : Frog-Man
  </h2>
  <h3
    data-testid="h3"
  >
     o tal vez no exista el personaje 
  </h3>
  <h4
    data-testid="h4"
  >
    o tal vez no alla elegido ningun personaje favorito tampoco
  </h4>
  <p
    data-testid="p"
  >
    Revise y vuelva a intentarlo por favor 
  </p>
</div>
`;
